torch/distributed/pipelining/PipelineSchedule.py

```diff
@@ -419,6 +419,12 @@ def _step_microbatches(
                 # forward
                 with record_function(f"Forward {i}"):
```

Comment:
Do you think it matters what string is written in the `record_function`?
If yes, maybe move the check to be above `record_function`
e.g. something like:
```
msgStr = "Forward {i}"
hasBackward = False
if i >= warmup_steps + 1 and self._has_backward:
   hasBackward = True
   msgStr = "Forward {i} with send back"
   with record_function(f"{msgStr}"):
      ops = self._stage.get_fwd_recv_ops()
      if hasBackward:
        # Note: coalesced bwd-send/fwd-recv
        # "i >= warmup_steps + 1" means "we ran a backward last iteration"
        # the backward send to prev stage should be coalesced with the fwd recv from the previous stage
        ops.extend(self._stage.get_bwd_send_ops())
  ```

Reply:
hm, yea we can use it fairly arbitrarily 

although long strings can be annoying, since this is what shows up in the profiler view.  But its not that long.  anyway, i will probably not mess with it in this PR.

i think it makes sense to holisitically improve logging in the schedule classes though.

specifically, i think we should move logging of send ops / recv ops _out_ of the stage class where the ops are created, and instead log them inside the schedule class where they are batched and sent.  cc @H-Huang @kwen2501